### DISCUSSION TITLE
ZY-M100-WIFI24G Occupancy vs. Status

### DISCUSSION THREAD

**Discussion by ingulacrist** [make-all/tuya-local#2645] [General]
I just read in the release notes of 2024.12.0 that _the deprecated status sensor has been removed from the ZYM100 and to use the motion and occupancy binary sensors instead_.

The thing is: only the _status_ sensor gives me any reliable occupancy detection.
I have 9 of these ZY-M100-WIFI24G sensors in the ceilings of my house and they all behave the same: _occupancy_ is basically random (will turn on in the middle of the night in a completely empty room) while _status_ only changes when somebody or something actually is in the room. 

The _status_ sensor also matches exactly what is displayed in the smart life app (_Nobody Status_) while _motion_ and _occupancy_ seem to be random and bear no connection to what is displayed in the app.

Was this behaviour fixed already or will the update to 2024.12.0 basically render my sensors useless? Also, am I the only one having this problem?

<img width="244" alt="image" src="https://github.com/user-attachments/assets/b6edde0a-3311-4a70-b395-4e9e2b7c2961" />
<img width="244" alt="image" src="https://github.com/user-attachments/assets/138b181c-ab1f-4b0d-9dcb-2473eacb4229" />

**Comment by make-all** [comment]
According to the details given in #1913, there are two dps that should give some form of "presence".
- dp 1 (previously used for Status and used for Motion sensor currently) indicates one of "presence", "move", or "none".
- dp 104 (used for Occupancy currently) indicates either "nobody" or "exist".

I assumed that the dp 104 was doing some higher level filtering to give a more reliable indication, but from your results that does not seem to be the case.  I just modified it so in 2024.12.2 and later, it should give the same behaviour as the old Status sensor for occupancy.

I'm not sure what you mean by "motion is random", as that is already indicating only the "move" state from the old Status sensor.

**Comment by ingulacrist** [comment]
Thanks for the reply. It's been a couple of months since I set it all up and motion was not a big focus as occupancy is my main use for the sensors. 
I just tested right now and _motion_ seems to be spot on.

I look forward to 2014.12.2, thanks.

**Reply by arsenicks** [reply]
Hey just wanted to drop a message to thank you for taking the time to report the issue. I had this on my never ending todo list to do a clean diagnostic et report an issue but never had enough time to find the issue and report it. FYI I had the exact same behaviour as you and started using the status sensor too while trying to understand what was the problem.

@make-all Thanks for the fix as far as I can tell right now the presence entity is working perfectly, will switch my automation to use that one now and while I'm at it, thanks for your work and time on this project I really appreciate it!



Have a good one!

**Comment by felipecrs** [comment]
As a user of the same device who upgraded to 2012.12.2 today and noticed the same, I'd like to leave my testimony about the previous behavior:

1. You should use the Motion sensor to trigger the scene start, when it turns on
2. You should the the Occupancy sensor to trigger the scene end, when it turns off

It was working flawlessly to me. The occupancy sensor is very sensitive, and must NOT be used for starting the scene. That's where the motion sensor comes in, which only detects bigger movements, like entering the room.

That said, I'll keep watching how it behaves on 2012.12.2. On a quick test, it seems to be working as well as before. Will let you know if I notice something strange.
